src/mcmc_gibbs.cpp

```diff
@@ -0,0 +1,138 @@
+#include "mcmc_gibbs.h"
+
+/*
+ * Helper function to iterate over constraints and apply them
+ */
+double add_constraint(const std::string& name, List constraints,
+                      IntegerVector districts, NumericVector &psi_vec,
```

Comment:
really like this solution to merging the two.

But can we make `districts` a `std::vector<int>` rather than an `IntegerVector`? The latter is a bit slower and gets allocated to the R memory pool, which can cause problems.

Reply:
Addressed in https://github.com/alarm-redist/redist/pull/121/commits/ad2dba5378ce1279b407aeb00ef6540d215e0876. Please give it a look, as I don't know this syntax super well.

Reply:
looks good, so as long as it tests ok for both flip and M-S then should be good to go for this part

Reply:
Great. Yeah, tests passed. mac still silly.